document/CoreFormat.mdk

```diff
@@ -2439,9 +2440,35 @@ C++: `"std::runtime_error"`
 C#: `"System.ArgumentNullException"`
 ~
 
+If the tool does not have access to information about the object that was thrown, the `type` property shall be absent.
+
+~ IsoNote
+This will be the case in C++ programs that throw objects that do not derive from `std::exception`:
```

Comment:
Yes, that's what I mean. If you own the catch site then of course you know the static type of the object -- it's the same name as the type you just typed. But the common exception scenario is that a piece of code catches some general exception like std::runtime_error or whatever, and the interesting part of what exploded is the dynamic type of the object. In C++ land, the way to get that information is via what(); the exception type itself is neither available nor typically very interesting.

Part of this too is that exceptions in C++ aren't used for "your program has a bug" kinds of situations -- you don't get an exception for trying to dereference a null pointer, for example, you get an immediate crash.